tests/FeedbackTest.php

```diff
@@ -21,6 +21,12 @@ protected function setUp() {
    * @covers Honeypot::getEncryptedTime
    */
   public function testHoneypotFieldsGenerated() {
+    $this->controller
+        ->shouldReceive('sendFeedback')
+        ->withAnyArgs()
+        ->once()
+        ->andReturn(true);
+
```

Comment:
Can your shortly explain the combination of methods used here, and why?

Reply:
This overrides the `sendFeedback` method of `$this->controller` (which is a partial mock class based on WebController) with a mock method that does nothing except returns `true`.

`withAnyArgs` means that any arguments are OK, the test will not fail no matter what was passed to the mock method.

`once` means that it is expected to be called exactly once; otherwise, the test will fail.

This is the same kind of mock that was already used in several other methods in this class, but it was previously forgotten for this method. What happened was that the original WebController::sendFeedback got called, and it tries to send an actual e-mail message, which fails since there is no recipient address. We don't want to call the true sendFeedback method in the test, just want to verify that it would have been called once.